validator/project_validator_test.go

```diff
@@ -49,47 +115,75 @@ buildvariants:
 
 		assert.Empty(t, errs)
 	})
-	t.Run("WarnsWithTaskDependingOnNonexistentTaskInSpecificBuildVariant", func(t *testing.T) {
+	t.Run("IgnoresDuplicateDependencies", func(t *testing.T) {
```

Comment:
The test changes from here down are just the splitting between `validateTaskNames` and `checkTaskNames`